StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet+API.swift

```diff
@@ -108,27 +117,35 @@ extension PaymentSheet {
                     }
                 } else {
                     let paymentIntentParams = confirmParams.makeParams(
-                        paymentIntentClientSecret: paymentIntent.clientSecret,
+                        paymentIntentClientSecret: clientSecret,
                         configuration: configuration
                     )
                     paymentIntentParams.returnURL = configuration.returnURL
-                    paymentIntentParams.shipping = makeShippingParams(for: paymentIntent, configuration: configuration)
+                    if let paymentIntent = paymentIntent {
+                        paymentIntentParams.shipping = makeShippingParams(
+                            for: paymentIntent,
+                            configuration: configuration
+                        )
+                    }
 
-                    // Paypal requires mandate_data if setting up
-                    if confirmParams.paymentMethodType.stpPaymentMethodType == .payPal
-                        && paymentIntent.setupFutureUsage == .offSession
-                    {
-                        paymentIntentParams.mandateData = .makeWithInferredValues()
+                    if let paymentIntent = paymentIntent {
```

Comment:
Besides shipping we also read SFU, we could read it from the IntentConfig and attach it to the `Intent.Confirmable`. Let me know your thoughts.